# Electric Vehicle Predictive Model

### Import dataset & show datahead

In [ ]:
import pandas as pd
import numpy as np
import scipy
from matplotlib import pyplot as plt
print ("Imported all libraries successfully...")

Imported all libraries successfully...


In [ ]:
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

data = pd.read_csv("Electric_Vehicle_Population_Data.csv")

In [ ]:
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import OneHotEncoder
import pandas as pd

features = data[['Make', 'Model Year', 'Electric Range', 'Electric Vehicle Type']]
target = data['Base MSRP']

features_encoded = pd.get_dummies(features, columns=['Make', 'Electric Vehicle Type'])

print(features_encoded.head())


   Model Year  Electric Range  Make_ALFA ROMEO  Make_AUDI  \
0        2016            14.0                0          0   
1        2018           215.0                0          0   
2        2014            38.0                0          0   
3        2021            32.0                0          0   
4        2018           215.0                0          0   

   Make_AZURE DYNAMICS  Make_BENTLEY  Make_BMW  Make_CADILLAC  Make_CHEVROLET  \
0                    0             0         1              0               0   
1                    0             0         0              0               0   
2                    0             0         0              0               1   
3                    0             0         0              0               0   
4                    0             0         0              0               0   

   Make_CHRYSLER  ...  Make_SMART  Make_SUBARU  Make_TESLA  Make_TH!NK  \
0              0  ...           0            0           0           0  

In [ ]:
# split data
X_train, X_test, y_train, y_test = train_test_split(features_encoded, target, test_size=0.2, random_state=42)

In [ ]:
# use random forest regressor
from sklearn.ensemble import RandomForestRegressor
model = RandomForestRegressor(n_estimators=100, random_state=42)

In [33]:
from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import mean_absolute_error

# Assuming you have already split the data and trained the model called 'model'
# Now, use the trained model to predict the MSRP on the test set
predictions = model.predict(X_test)

# Calculate the Mean Absolute Error between the actual and predicted MSRP values
mae = mean_absolute_error(y_test, predictions)

# Output the MAE to understand the average deviation of the model's predictions from actual prices
print(f"Mean Absolute Error (MAE) on test data: {mae}")


Mean Absolute Error (MAE) on test data: 110.75666916129502


The MAE of 110.76 means that, on average, the model's predictions of electric vehicle MSRPs are off by about $110.76 from their actual values.

In [32]:
# Assuming 'df' is your training dataframe and 'model' is your trained model

# Preprocess the new data the same way as the training data
new_vehicle_data = {
    'Model Year': [2015],
    'Electric Range': [250],
    'Make': ['Tesla'],  # Using 'Make' column before one-hot encoding
    'Electric Vehicle Type': ['Battery Electric Vehicle (BEV)']
}

# Convert to DataFrame
new_vehicle_df = pd.DataFrame(new_vehicle_data)

# Apply one-hot encoding to match training data structure
new_vehicle_df_encoded = pd.get_dummies(new_vehicle_df)

# Ensure all training features are present, filling missing ones with 0s
final_vehicle_df = new_vehicle_df_encoded.reindex(columns=model.feature_names_in_, fill_value=0)

# Predict the MSRP using the trained model
predicted_msrp = model.predict(final_vehicle_df)

print(f"Predicted MSRP: ${predicted_msrp[0]:,.2f}")


Predicted MSRP: $0.00
